cluster-install/install.sh

```diff
@@ -1369,6 +1369,33 @@ create_kubectl_patch_for_containers() {
     _return_value="$resource -n $namespace --type=json -p '$json_patch'"
 }
 
+backup_kubernetes_resource() {
+    local -r kind="$1"
+    local -r resource_name="$2"
+    local -r directory="$3"
+    local -r namespace="${4:-""}"
+
+    [ -z "$kind" ] && fatal "no kind given"
+    [ -z "$resource_name" ] && fatal "no resource_name given"
+    [ -z "$directory" ] && fatal "no directory given"
+    ! [ -d "$directory" ] && fatal "directory '$directory' does not exist"
+
+    local namespace_arg=""
+    local backup_name="BACKUP_"
+    if [ -n "$namespace" ]; then
+        namespace_arg="--namespace='$namespace'"
+        backup_name+="${namespace}_"
+    fi
+    backup_name+="${kind}_${resource_name}.yaml"
+
+    local -r resource_name_yaml="$(kubectl get "$kind" "$resource_name" "$namespace_arg" -o yaml 2> $__ERR_FILE)"
+    [ -z "$resource_name_yaml" ] && return 1
+    _return_error="$(get_captured_err)"
+
+    echo "$resource_name_yaml" > "${directory}/${backup_name}"
+    echo "$backup_name"
+}
```

Comment:
This backup function is used to backup the TORC before/if we apply patches to it.